src/nanoarrow/nanoarrow_testing_test.cc

```diff
@@ -1107,3 +1147,290 @@ TEST(NanoarrowTestingTest, NanoarrowTestingTestFieldUnion) {
   TestTypeError(R"({"name": "union", "mode": "NOT_A_MODE", "typeIds": []})",
                 "Type[name=='union'] mode must be 'DENSE' or 'SPARSE'");
 }
+
+TEST(NanoarrowTestingTest, NanoarrowTestingTestSchemaComparison) {
+  nanoarrow::UniqueSchema actual;
+  nanoarrow::UniqueSchema expected;
+  TestingJSONComparison comparison;
+  std::stringstream msg;
+
+  // Start with two identical schemas and ensure there are no differences
+  ArrowSchemaInit(actual.get());
+  ASSERT_EQ(ArrowSchemaSetTypeStruct(actual.get(), 1), NANOARROW_OK);
+  ASSERT_EQ(ArrowSchemaSetType(actual->children[0], NANOARROW_TYPE_NA), NANOARROW_OK);
+  ASSERT_EQ(ArrowSchemaDeepCopy(actual.get(), expected.get()), NANOARROW_OK);
+
+  ASSERT_EQ(comparison.CompareSchema(actual.get(), expected.get()), NANOARROW_OK);
+  EXPECT_EQ(comparison.num_differences(), 0);
+  comparison.WriteDifferences(msg);
+  EXPECT_EQ(msg.str(), "");
+  msg.str("");
+  comparison.ClearDifferences();
```

Comment:
You could factor this out in a helper function:
```suggestion
  AssertSchemasCompareEqual(actual.get(), expected.get());
```